apps/console/src/features/roles/components/role-list.tsx

```diff
@@ -232,23 +232,16 @@ export const RoleList: React.FunctionComponent<RoleListProps> = (props: RoleList
                 id: "audience",
                 key: "audience",
                 render: (role: RolesInterface) => (
-                    <Header as="h6" data-componentid={ `${ componentId }-col-2-item-heading` }>
-                        <Header.Content>
-                            <Header.Subheader data-componentid={ `${ componentId }-col-2-item-sub-heading` }>
-                                { role?.audience?.type.charAt(0).toUpperCase() + role?.audience?.type.slice(1) }
-                                {
-                                    RoleAudienceTypes.APPLICATION === role?.audience?.type.toUpperCase() && (
-                                        <Label
-                                            size="mini"
-                                            className="client-id-label"
-                                        >
-                                            { role?.audience?.display }
-                                        </Label>
-                                    )
-                                }
-                            </Header.Subheader>
-                        </Header.Content>
-                    </Header>
+                    role?.audience && (
```

Comment:
Can we guarantee that audience always include `type` and `display` properties?

If yes, checking `role?.audience` would be enough.

Otherwise, we need to modify the condition to something like `role?.audience?.type && role?.audience?.display`.

Reply:
`type` and `display` are mandatory values  of `RoleAudiencesInterface.

Reply:
Typescript cannot check the presence of object properties in runtime. Therefore, if there is no backend logic to validate these properties are inserted to role, we must use `role?.audience?.type && role?.audience?.display` otherwise, the app would render the `something went wrong` error or display audience as `undefined`